provider/kubernetes/kubernetes.go

```diff
@@ -211,8 +214,21 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 						ruleType = ruleTypePathPrefix
 					}
 
-					templateObjects.Frontends[r.Host+pa.Path].Routes[pa.Path] = types.Route{
-						Rule: ruleType + ":" + pa.Path,
+					rewriteTarget := i.Annotations[annotationKubernetesRewriteTarget]
+					if len(rewriteTarget) > 0 {
+						ruleType = ruleTypeReplacePath
+					}
+
+					switch {
+					case ruleType == ruleTypeReplacePath:
+						templateObjects.Frontends[r.Host+pa.Path].Routes[pa.Path] = types.Route{
+							Rule: ruleType + ":" + rewriteTarget,
+						}
+
+					default:
+						templateObjects.Frontends[r.Host+pa.Path].Routes[pa.Path] = types.Route{
+							Rule: ruleType + ":" + pa.Path,
+						}
```

Comment:
Could we move the logic pertaining to setting the route type to a dedicated function to ease readability?